components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/main/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/services/MessageFlowReporterThread.java

```diff
@@ -267,12 +268,26 @@ void addFaultsToParents(List<StatisticsLog> messageFlowLogs, BasicStatisticDataU
             StatisticsLog updatingLog = messageFlowLogs.get(index);
             updatingLog.incrementNoOfFaults();
             index = updatingLog.getParentIndex();
+            setElasticMetaData(dataUnit, updatingLog);
+        }
+
+    }
 
-            if (dataUnit.getElasticMetadata() != null) {
-                updatingLog.setElasticMetadata(dataUnit.getElasticMetadata());
+    private void setElasticMetaData(BasicStatisticDataUnit dataUnit, StatisticsLog statisticsLog) {
+        if (dataUnit.getElasticMetadata() == null) {
+            dataUnit.generateElasticMetadata(dataUnit.getMessageContext());
+        }
+        ElasticMetadata elasticMetadata = dataUnit.getElasticMetadata();
+        AnalyticsCustomDataProvider customDataProvider = AnalyticsDataProviderHolder.getAnalyticsCustomDataProvider();
+        if (customDataProvider != null) {
+            if (elasticMetadata.getAnalyticsMetadata() == null) {
+                dataUnit.getMessageContext().setProperty(ANALYTICS_METADATA, new HashMap<String, Object>());
```

Comment:
Can dataunit message context null?

Reply:
From the synapse side, it cannot be. But since the Data Unit is a public class which can be initialized elsewhere, will add a null check

Reply:
Fixed with 238fadfeede47fcf1733dea02bed46be6aefe20e